src/vegas_mode/plugin_adapter.py

```
@@ -0,0 +1,368 @@
+            List with single captured image, or None
+        """
+        original_image = None
+        try:
+            # Save current display state
+            original_image = self.display_manager.image.copy()
+
+            # Ensure plugin has fresh data before capturing
+            if hasattr(plugin, 'update_data'):
+                try:
+                    plugin.update_data()
+                except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)